src/libaktualizr/http/httpclient.cc

```diff
@@ -35,8 +35,14 @@ static size_t writeString(void* contents, size_t size, size_t nmemb, void* userp
   return size * nmemb;
 }
 
+std::mutex HttpClient::curl_init_mutex;
+
 HttpClient::HttpClient(const std::vector<std::string>* extra_headers) {
-  curl = curl_easy_init();
+  {
```

Comment:
Is it necessary here? If `curl_global_init()` was called before, `curl_easy_init()` should be thread safe: https://curl.haxx.se/libcurl/c/threadsafe.html